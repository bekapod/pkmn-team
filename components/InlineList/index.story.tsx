// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import InlineList from ".";

storiesOf("InlineList", module).add("default", () => (
  <InlineList>
    <li>Item 1</li>
    <li>Item 2</li>
  </InlineList>
));
