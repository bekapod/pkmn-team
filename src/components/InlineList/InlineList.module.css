.base {
  display: flex;
  flex-wrap: wrap;
  margin: 0;
  padding: 0;
  list-style: none;

  & > *:not(:last-child) {
    margin-right: var(--spacing-3);
  }
}
