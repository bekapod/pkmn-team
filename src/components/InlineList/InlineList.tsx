import type { FunctionComponent, ComponentPropsWithRef } from 'react';
import cx from 'classnames';
import styles from './InlineList.module.css';

export const InlineList: FunctionComponent<ComponentPropsWithRef<'ul'>> = ({
  className,
  ...props
}) => <ul className={cx(styles.base, className)} {...props} />;
