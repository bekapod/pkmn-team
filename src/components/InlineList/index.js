// @flow
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

const InlineList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  margin: 0;
  padding: 0;
  list-style: none;

  & > * {
    margin-right: ${variables.spacing.sm}px;
  }
`;

export default InlineList;
