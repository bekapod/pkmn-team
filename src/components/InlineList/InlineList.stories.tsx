import { Meta, Story } from '@storybook/react/types-6-0';
import { InlineList } from './InlineList';

export default {
  title: 'Components/Inline List',
  component: InlineList
} as Meta;

export const inlineList: Story = () => (
  <InlineList>
    <li>Item 1</li>
    <li>Item 2</li>
  </InlineList>
);
