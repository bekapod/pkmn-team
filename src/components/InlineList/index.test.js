// @flow
import React from "react";
import renderer from "react-test-renderer";
import InlineList from ".";

describe("<InlineList />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(
      <InlineList>
        <li>Item 1</li>
        <li>Item 2</li>
      </InlineList>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
