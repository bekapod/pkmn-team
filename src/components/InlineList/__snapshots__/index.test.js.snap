// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineList /> renders without crashing 1`] = `
<ul
  className="InlineList-bwsyp4-0 bKIwJu"
>
  <li>
    Item 1
  </li>
  <li>
    Item 2
  </li>
</ul>
`;
